src/Config/Option/BooleanOption.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+/**
+ * ---------------------------------------------------------------------
+ *
+ * GLPI - Gestionnaire Libre de Parc Informatique
+ *
+ * http://glpi-project.org
+ *
+ * @copyright 2015-2024 Teclib' and contributors.
+ * @copyright 2003-2014 by the INDEPNET Development Team.
+ * @licence   https://www.gnu.org/licenses/gpl-3.0.html
+ *
+ * ---------------------------------------------------------------------
+ *
+ * LICENSE
+ *
+ * This file is part of GLPI.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * ---------------------------------------------------------------------
+ */
+
+namespace Glpi\Config\Option;
+
+use Glpi\Application\View\TemplateRenderer;
+use Glpi\Config\ConfigOption;
+use Glpi\Config\ConfigScope;
+use Glpi\Config\ConfigSection;
+use Glpi\Config\InputType;
+
+class BooleanOption extends ConfigOption
+{
+    public function __construct(
+        array $scopes,
+        ConfigSection $section,
+        string $name,
+        string $label,
+        InputType $input_type = InputType::CHECKBOX,
```

Comment:
Slider is used at least 7 times and checkbox at least 8 in `templates/pages/setup` alone.  This PR is already atrociously big for only having the options from the General tab of Setup > General and user preferences. It would be a shame if we cannot declare a configuration option once and have it available in both the advanced config UI and the regular location.